src/Microsoft.DotNet.Build.Tasks/VersionTools/LocalUpdatePublishedVersions.cs

```diff
@@ -5,31 +5,61 @@
 using Microsoft.Build.Framework;
 using Microsoft.DotNet.VersionTools.Automation;
 using System.Diagnostics;
-using System.Linq;
 
 namespace Microsoft.DotNet.Build.Tasks.VersionTools
 {
     public class LocalUpdatePublishedVersions : BuildTask
     {
         [Required]
-        public ITaskItem[] ShippedNuGetPackage { get; set; }
+        public string[] ShippedNuGetPackage { get; set; }
 
         [Required]
         public string VersionsRepoLocalBaseDir { get; set; }
 
         [Required]
         public string VersionsRepoPath { get; set; }
 
+        public string GitHubAuthToken { get; set; }
+        public string GitHubUser { get; set; }
+
+        /// <summary>
+        /// If specified, create the local build-infos based on the information available in the
+        /// versions repo. Specifically, Latest_Packages.txt will contain the latest version of each
+        /// package, even if this build didn't produce that certain package. Useful when servicing,
+        /// where a subset of packages are built.
+        /// </summary>
+        public string VersionsRepo { get; set; }
+        public string VersionsRepoOwner { get; set; }
+        public string VersionsRepoBranch { get; set; } = "master";
+
         public override bool Execute()
         {
             Trace.Listeners.MsBuildListenedInvoke(Log, () =>
             {
                 var updater = new LocalVersionsRepoUpdater();
 
-                updater.UpdateBuildInfoLatestPackages(
-                    ShippedNuGetPackage.Select(item => item.ItemSpec),
-                    VersionsRepoLocalBaseDir,
-                    VersionsRepoPath);
+                if (!string.IsNullOrEmpty(GitHubAuthToken))
+                {
+                    updater.GitHubAuth = new GitHubAuth(GitHubAuthToken, GitHubUser);
+                }
+
+                GitHubBranch branch = null;
+                if (!string.IsNullOrEmpty(VersionsRepo))
```

Comment:
In what case would the versions repo element be non-empty for a local repo?

Reply:
When this is used by prodcon, it'll be set to `versions` so that the task creates both Latest_Packages.txt and Last_Build_Packages.txt (where Latest_Packages.txt preserves the info we want for servicing scenarios).

When this is used by source-build, it's null.